File: csharp/src/code-smells/bloaters/PrimitiveObsession.cs
namespace RefactoringAvanzado.CodeSmells.Bloaters;

public class Order
{
    private readonly string customerName;
    private readonly string customerEmail;
    private readonly string address;
    private readonly double totalAmount;
    private readonly string currency;

    public Order(
        string customerName,
        string customerEmail,
        string address,
        double totalAmount,
        string currency)
    {
        this.customerName = customerName;
        this.customerEmail = customerEmail;
        this.address = address;
        this.totalAmount = totalAmount;
        this.currency = currency;
    }

    public void SendInvoice()
    {
        if (!customerEmail.Contains("@"))
        {
            throw new Exception("Email inválido");
        }
        if (string.IsNullOrEmpty(address))
        {
            throw new Exception("No se ha indicado dirección");
        }
        if (totalAmount <= 0)
        {
            throw new Exception("El monto debe ser mayor que cero");
        }
        Console.WriteLine($"Factura enviada a {customerName} in {address} por {totalAmount} {currency}");
    }
}

File: typescript/src/code-smells/bloaters/primitive-obsession.ts
class Order {
  constructor(
    private customerName: string,
    private customerEmail: string,
    private address: string,
    private totalAmount: number,
    private currency: string,
  ) {
  }

  sendInvoice() {
    if (!this.customerEmail.includes('@')) {
      throw new Error('Email inválido')
    }
    if (!this.address) {
      throw new Error('No se ha indicado dirección')
    }
    if (this.totalAmount <= 0) {
      throw new Error('El monto debe ser mayor que cero')
    }
    console.log(`Factura enviada a ${this.customerName} in ${this.address} por ${this.totalAmount} ${this.currency}`)
  }
}
